new_platforms/docs/new_platform_dev.md

```diff
@@ -1,392 +1,462 @@
-so under both Windows and Linux. Again, the Open Enclave SDK deals with these
-details on your behalf. The `SampleServerApp` folder follows the same
-principles.
+The `enc` directory holds the source code for your enclave, in this case a
+single `enc.c` file. Under the `optee` folder there are files that are used by
+the build process for OP-TEE. Currently, OP-TEE requires building TAs with GNU
+Make and does not support CMake. The `sub.mk` file specifies the list of sources
+a second time as well as per-file compiler flags, where necessary. Similarly,
+the `linux_gcc.mak` file also contains a copy of the UUID value. The
+`user_ta_header_defines.h` header file contains configuration values required by
+OP-TEE, including the UUID as well in a C-structure format. The `uuids.reg`
+registry file contains registry keys and values to register an OP-TEE TA with
+the Windows Trusted Runtime via its UUID.
+
+The file `myproject_enclave_private.pem` holds the private key in PEM format
+with which to sign the enclave when the target TEE is Intel SGX. The file
+`myproject_enclave_config.xml` contains configuration values required by Intel
+SGX such as the max stack and heap sizes.
+
+Lastly, there is a `CMakeLists.txt` file that describes to CMake how to build
+the enclave.
+
+The `host` directory holds the code for your host app, in this case a single
+`main.c` file. This file contains OS- and TEE-agnostic host code that loads the
+TA. Note that the code that you write for your host app need not know how to
+launch an enclave under Intel SGX or a TA under OP-TEE nor how to do so under
+both Windows and Linux. The Open Enclave SDK deals with these details on your
+behalf. The `host` directory also has a `CMakeLists.txt` file that instructs
+CMake how to build the host app executable.
 
 In general then, there is an EDL file that both the trusted and untrusted
 components consume that specifies how the pair communicate across the trust
-boundary. The EDL file only contains functions that are specific to your
-use case. The trusted component need only be written once and targets the Open
+boundary. The EDL file only contains functions that are specific to your use
+case. The trusted component need only be written once and targets the Open
 Enclave SDK's API. This way it can be seamlessly compiled as an Intel SGX
-enclave and as an ARM TrustZone trusted application. Similarly, the host
-programs also need only be written once and, assuming you do not make use of
-OS-specific functionality, they too can be compiled into Windows and Linux
-programs that can seamlessly launch and operate Intel SGX enclaves and ARM
-TrustZone trusted applications.
+enclave and as an OP-TEE TA. Similarly, the host programs also need only be
+written once and, assuming you do not make use of OS-specific functionality,
+they too can be compiled into Windows and Linux programs that can seamlessly
+launch and operate Intel SGX enclaves and OP-TEE TAs.
+
+## Building the Enclave
```

Comment:
This should be in a section (or separate page) about using cmake instead of Visual Studio.
We need to keep the section (or page) on using Visual Studio for apps.